tests/integration/test_post_endpoint.py

```diff
@@ -0,0 +1,146 @@
+from encryption_key_store.core.db_interop.mongo_dao import MongoDbDao
+
+from ..fixtures.file_fixture import first_part_fixture  # noqa: F401
+from ..fixtures.file_fixture import FirstPartFixture
+from ..fixtures.ghga_secrets import generate_secrets_fixture  # noqa: F401
+from ..fixtures.ghga_secrets import ghga_secrets_dao_fixture  # noqa: F401
+
+client = TestClient(app=app)
+
+
+@pytest.mark.asyncio
+async def test_post_secrets(
+    *,
+    first_part_fixture: FirstPartFixture,  # noqa: F811
+):
+    """Test request response for /secrets endpoint with valid data"""
+
+    async def dao_override() -> MongoDbDao:
+        """Ad hoc DAO dependency overridde"""
+        return first_part_fixture.ghga_secrets_dao_fixture.dao
+
+    app.dependency_overrides[dao_injector] = dao_override
+
+    payload = first_part_fixture.content
+
+    request_body = {
+        "user_id": "Test-User",
+        "file_part": base64.b64encode(payload).hex(),
+    }
+    response = client.post(url="/secrets", json=request_body)
+    assert response.status_code == 200
+    body = response.json()
+    secret = base64.b64decode(codecs.decode(body["secret"], "hex"))
+    assert secret
+    assert body["secret_id"]
+    assert body["offset"] > 0
+
+
+@pytest.mark.asyncio
+async def test_corrupted_header(
+    *,
+    first_part_fixture: FirstPartFixture,  # noqa: F811
+):
+    """Test request response for /secrets endpoint with first char replaced in envelope"""
+
+    async def dao_override() -> MongoDbDao:
+        """Ad hoc DAO dependency overridde"""
+        return first_part_fixture.ghga_secrets_dao_fixture.dao
+
+    app.dependency_overrides[dao_injector] = dao_override
+
+    payload = b"k" + first_part_fixture.content[2:]
+    content = base64.b64encode(payload).hex()
+
+    request_body = {
+        "user_id": "Test-User",
+        "file_part": content,
+    }
+    response = client.post(url="/secrets", json=request_body)
+    assert response.status_code == 400
+    body = response.json()
+    assert body["exception_id"] == "malformedOrMissingEnvelopeError"
+
+
+@pytest.mark.asyncio
+async def test_missing_envelope(
+    *,
+    first_part_fixture: FirstPartFixture,  # noqa: F811
+):
+    """Test request response for /secrets endpoint without envelope"""
+
+    async def dao_override() -> MongoDbDao:
+        """Ad hoc DAO dependency overridde"""
+        return first_part_fixture.ghga_secrets_dao_fixture.dao
+
+    app.dependency_overrides[dao_injector] = dao_override
+
+    payload = first_part_fixture.content
+    content = base64.b64encode(payload).hex()
+    content = content[124:]
+
+    request_body = {
+        "user_id": "Test-User",
+        "file_part": content,
+    }
+    response = client.post(url="/secrets", json=request_body)
+    assert response.status_code == 400
+    body = response.json()
+    assert body["exception_id"] == "malformedOrMissingEnvelopeError"
+
+
+@pytest.mark.asyncio
+async def test_invalid_secret(
```

Comment:
To avoid confusions, maybe it would bet helpful to only use the term 'secret' for the symmetric file secret. Crypt4GH secrets can be referred to as Crypt4GH private key.